src/object.js

```diff
@@ -469,38 +451,56 @@ module.exports = (common) => {
 
           ipfs.object.put(testObj, (err, node) => {
             expect(err).to.not.exist
-            node.multihash((err, multihash) => {
-              expect(err).to.not.exist
 
-              ipfs.object.links(multihash, (err, links) => {
-                expect(err).to.not.exist
-                expect(node.links).to.deep.equal(links)
-                done()
-              })
+            ipfs.object.links(node.multihash, (err, links) => {
+              expect(err).to.not.exist
+              expect(node.links).to.deep.equal(links)
+              done()
             })
           })
         })
 
         it('with multihash (+ links)', (done) => {
-          const node1 = new DAGNode(new Buffer('Some data 1'))
-          const node2 = new DAGNode(new Buffer('Some data 2'))
-
-          node1.addNodeLink('some-link', node2, (err) => {
-            expect(err).to.not.exist
+          let node1a
+          let node1b
+          let node2
 
-            ipfs.object.put(node1, (err, node) => {
-              expect(err).to.not.exist
-              node.multihash((err, multihash) => {
+          series([
+            (cb) => {
+              DAGNode.create(new Buffer('Some data 1'), (err, node) => {
+                expect(err).to.not.exist
+                node1a = node
+                cb()
+              })
+            },
+            (cb) => {
+              DAGNode.create(new Buffer('Some data 2'), (err, node) => {
                 expect(err).to.not.exist
+                node2 = node
+                cb()
+              })
+            },
```

Comment:
Same as in https://github.com/ipld/js-ipld-resolver/pull/67#discussion_r89480770